idaes/power_generation/unit_models/drum1D.py

```diff
@@ -0,0 +1,1171 @@
+        self.number_downcomer = Var(
+                initialize=4,
+                doc="Number of Downcomers Connected to Drum")
+
+        # Inside diameter of downcomer
+        self.downcomer_di = Var(
+                initialize=0.6,
+                doc="Inside Diameter of Downcomer")
+
+        # Inside Radius expression
+        @self.Expression(doc="Inside Radius of Drum")
+        def drum_ri(b):
+            return 0.5*b.drum_di
+
+        # Outside Radius expression
+        @self.Expression(doc="Outside radius of drum")
+        def drum_ro(b):
+            return b.drum_ri + b.drum_thickness
+
+        # Outside diameter expression
+        @self.Expression(doc="Outside Radius of Drum")
+        def drum_do(b):
+            return b.drum_di + 2 * b.drum_thickness
+
+        # Inner surface area (ignore two hemispheres at ends)
+        @self.Expression(doc="Inner Surface Area")
+        def drum_area(b):
+            return const.pi*b.drum_di*b.drum_length
+
+    def _make_performance(self):
+        """
+        Define constraints which describe the behaviour of the unit model.
+        """
+
+        # Thermal conductivity of drum
+        self.therm_cond_metal = Param(initialize=40, mutable=True)
+
+        # Thermal conductivity of insulation
+        self.therm_cond_insulation = Param(initialize=0.08, mutable=True)
+
+        # Thermal conductivity of air
+        self.therm_cond_air = Param(initialize=0.03, mutable=True)
+
+        # Density of drum material
+        self.dens_metal = Param(initialize=7753, mutable=True)
+
+        # Heat capacity of drum material
+        self.cp_metal = Param(initialize=486, mutable=True)
+
+        # Young modulus
+        self.Young_modulus = Param(initialize=2.07E5, mutable=True)
+
+        # Poisson's ratio
+        self.Poisson_ratio = Param(initialize=0.292, mutable=True)
+
+        # Coefficient of thermal expansion
+        self.coefficient_therm_expansion = Param(initialize=1.4E-5,
+                                                 mutable=True)
+
+        # Constant related to Ra,
+        # (gravity*expansion_coefficient*density
+        # /viscosity/thermal_diffusivity)^(1/6)
+        # Use properties at 50 C and 1 atm, 6.84e7^(1/6)=20.223
+        self.const_Ra_root6 = Param(initialize=20.223, mutable=True)
+
+        # Constant related to Nu for free convection,
+        # 0.387/(1+0.721*Pr^(-9/16))^(8/27)
+        # Use properties at 50 C and 1 atm
+        self.const_Nu = Param(initialize=0.322, mutable=True)
+
+        # Ambient pressure
+        self.pres_amb = Param(initialize=1.0E5, doc="Ambient Pressure")
+
+        # Ambient temperature
+        self.temperature_ambient = Var(self.flowsheet().config.time,
+                                       initialize=298.15,
+                                       doc="Ambient Temperature")
+
+        # Inside heat transfer coefficient
+        self.h_in = Var(self.flowsheet().config.time, initialize=1,
```

Comment:
I think we should use a more descriptive name here, as "h" can mean many things in process engineering (e.g. specific enthalpy).